tests/unit-tests/test_trainer.py

```diff
@@ -518,6 +519,10 @@ def compare_results(target_res, check_res):
                     assert_equal(tr_1, cr_1)
                     assert_equal(tr_2, cr_2)
                 else:
+                    # In case LP results also returned candidate list
+                    # lengths, we only keep the rankings
+                    if isinstance(check_r, tuple):
+                        check_r, _ = check_r
```

Comment:
Shall we also check the lengths?

Reply:
Check the lengths to ensure rankings and batch lengths have the correct (same) lengths?